uma/uma.go

```diff
@@ -1011,3 +1020,64 @@ func VerifyPostTransactionCallbackSignature(
 	}
 	return verifySignature(*signablePayload, *callback.Signature, otherVaspPubKeyResponse)
 }
+
+func CreateUmaInvoice(
```

Comment:
I know we did things this way elsewhere in the uma go SDK, but I'm wondering if there are better object creation conventions in Go (I didn't really know Go well when I was making this sdk). For example, I've noticed that most libraries use the New* naming convention, like `NewUmaInvoice`. I'm also wondering if it would make sense to have an options struct for the optional args, kind of [like this article](https://medium.com/star-gazers/go-options-pattern-da49185a2526) suggests. Totally optional suggestions, but figured it'd be good to see if we can do better this time from the start. I kind of regret how verbose and ugly some parts the go uma sdk are :-/... soooo many params lol.

Reply:
This one has to create an unsigned invoice and sign it. So I think we should stick with this method. However for other objects, we could utilize New